cmd/soroban-cli/src/commands/contract/info/shared.rs

```diff
@@ -84,18 +113,26 @@ pub async fn fetch_wasm(args: &Args) -> Result<Option<Vec<u8>>, Error> {
             .verify_network_passphrase(Some(&network.network_passphrase))
             .await?;
 
-        get_remote_wasm_from_hash(&client, &hash).await?
+        print.globeln(format!(
+            "Downloading contract spec for wasm hash: {wasm_hash}"
+        ));
+        Ok((
+            Some(get_remote_wasm_from_hash(&client, &hash).await?),
+            None,
+            Some(network.clone()),
+        ))
     } else if let Some(contract_id) = &args.contract_id {
         let contract_id =
             contract_id.resolve_contract_id(&args.locator, &network.network_passphrase)?;
+        let contract_address = xdr::ScAddress::Contract(xdr::Hash(contract_id.0)).to_string();
+        print.globeln(format!("Downloading contract spec: {contract_address}"));
         let res = wasm::fetch_from_contract(&contract_id, network).await;
         if let Some(ContractIsStellarAsset) = res.as_ref().err() {
-            return Ok(None);
+            print.globeln("Contract is a Stellar asset; it has no spec");
```

Comment:
Should it just return an error instead?

Reply:
We need to return `None` so that the consumer can do the right thing here. For example, the `contract info meta` command will throw with "Stellar asset contract doesn't contain meta information" in the case of SAC.

Reply:
(Also, was trying to keep existing `contract info` behavior unchanged!)

Reply:
So basically, `None` is returned here so that we can throw/return an error in the consumer.

BUT, thanks for asking about this! Now that I looked into it more, I noticed that we were throwing an error in the `bindings typescript` case, thinking we could not get a spec for the SAC. But `contract info interface` showed me how to generate a SAC spec! So `bindings typescript` no longer needs to throw an error in the `None` case here, and there's no reason to log anything here. 💪🏼